### PR TITLE
Feature/codeclimate fixes equals error

### PR DESCRIPTION
**Pull request by niekraaijmakers**

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java [hunk 1]
```diff
@@ -37,6 +37,9 @@
 import java.util.Map;
 import java.util.UUID;
 
+import static org.apache.commons.lang3.ArrayUtils.contains;
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java]: static import of methods is unclear and makes it appear these methods are implemented by this class. better to clearly qualify where they come from in code.

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java [hunk 2]
```diff
@@ -39,6 +39,9 @@
 import java.util.Map;
 import java.util.UUID;
 
+import static org.apache.commons.collections.CollectionUtils.containsAny;
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java]: static import of methods is unclear and makes it appear these methods are implemented by this class. better to clearly qualify where they come from in code.

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ResourcePropertiesHttpCacheConfigExtension.java [hunk 3]
```diff
@@ -35,7 +35,14 @@
 import org.osgi.service.metatype.annotations.Designate;
 import org.osgi.service.metatype.annotations.ObjectClassDefinition;
 
-import java.util.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.apache.commons.collections.CollectionUtils.containsAny;
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ResourcePropertiesHttpCacheConfigExtension.java]: static import of methods is unclear and makes it appear these methods are implemented by this class. better to clearly qualify where they come from in code.

**Comment by niekraaijmakers** [comment]
Good question regarding the map to hashmap. Because map isnt serializeable it generates a codeclimate warning. I could also suppress the warning alternatively?

**Comment by justinedelson** [comment]
> Good question regarding the map to hashmap. Because map isnt serializeable it generates a codeclimate warning. I could also suppress the warning alternatively?

If I understand the warning emitted by CodeClimate, it has more to do with the serialization.

My suggestion would be to leave `AbstractKeyValueExtension` as is (with `Map`) and then create a new HashMap when serializing.

**Comment by niekraaijmakers** [comment]
I guess that can work as well, so wrap the map with a hashmap on serialization. I'll do that one. Give me some time as I'm quite busy with 2 projects :)

**Comment by niekraaijmakers** [comment]
thanks !! @davidjgonzalez  do you approve sir?

**Pull request merged**